src/psyclone/psyGen.py

```diff
@@ -2131,8 +2170,20 @@ def __init__(self, field, check_dirty=True,
         self._halo_depth = None
         self._check_dirty = check_dirty
         self._vector_index = vector_index
-        self._text_name = "HaloExchange"
-        self._colour_key = "HaloExchange"
+
+    @staticmethod
```

Comment:
I don't like the subclassing here as leaf statements can appear in different places of the current class hierarchy and we would need multiple inheritance for it. But I made the generic implementation behave like a Leaf node. (I think in this case is it worth breaking the OO rule of expansion>>>modification since the non-leaf nodes have to always specialize its own validate method anyway)

The effect is the same, I could delete all duplicated Leaf node validations.